segno/encoder.py

```diff
@@ -1082,7 +1082,7 @@ def make_segment(data, mode, encoding=None):
         if _PY2:  # pragma: no cover
             segment_data = [ord(b) for b in segment_data]
         # Note: len(segment.data)! segment.data_length = len(segment.data) / 2!!
-        for i in range(0, char_count, 2):
+        for i in range(0, len(segment_data), 2):
```

Comment:
After encoding, 1 character is mapped to 2 elements in `segment_data`. Here we should use `len(segment_data)` as the upper bound. Otherwise only half data is added to segment.
We'd better use zxing to scan the generated code as a **test** so that we can discover these bugs.

Reply:
Thanks again, fixed in 1.0.2
Added test cases which use zbar to decode the generated QR Codes.

Hanzi: Thanks! I'll take a look at it, may take till begin of next week, though.